components/swim/src/member.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2016 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fmt;
+use std::iter::IntoIterator;
+use std::net::SocketAddr;
+use std::ops::{Deref, DerefMut};
+
+use uuid::Uuid;
+use rand::{thread_rng, Rng};
+
+use message::swim::Member as ProtoMember;
+
+const PINGREQ_TARGETS: usize = 5;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Health {
+    Alive,
+    Suspect,
+    Confirmed,
+}
+
+impl fmt::Display for Health {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &Health::Alive => write!(f, "Alive"),
+            &Health::Suspect => write!(f, "Suspect"),
+            &Health::Confirmed => write!(f, "Confirmed"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Member {
+    pub proto: ProtoMember,
+}
+
+impl Member {
+    pub fn new() -> Member {
+        let mut proto_member = ProtoMember::new();
+        proto_member.set_id(Uuid::new_v4().simple().to_string());
+        proto_member.set_incarnation(0);
+        Member{
+            proto: proto_member,
+        }
+    }
+
+    pub fn new_from_proto(proto: ProtoMember) -> Member {
```

Comment:
This function is just fine, but the `Into<T>` trait may be the more "rust appropriate" way to do it. That would let you call `proto.into()` instead of `Member::new_from_proto(proto)`